src/m365/spo/commands/web/web-applicationcustomizer-add.ts

```diff
@@ -0,0 +1,105 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { validation } from '../../../../utils/validation';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import { Options as spoCustomActionAddCommandOptions } from '../customaction/customaction-add';
+import * as spoCustomActionAddCommand from '../customaction/customaction-add';
+import Command from '../../../../Command';
+import { Cli } from '../../../../cli/Cli';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  webUrl: string;
+  clientSideComponentId: string;
+  clientSideComponentProperties?: string;
+}
+
+class SpoWebApplicationCustomizerAddCommand extends SpoCommand {
+  public get name(): string {
+    return commands.WEB_APPLICATIONCUSTOMIZER_ADD;
+  }
+
+  public get description(): string {
+    return 'Add an application customizer to a site.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-t, --title <title>'
+      },
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --clientSideComponentId <clientSideComponentId>'
+      },
+      {
+        option: '--clientSideComponentProperties [clientSideComponentProperties]'
+      },
+    );
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        clientSideComponentProperties: typeof args.options.clientSideComponentProperties !== 'undefined'
+      });
+    });
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
```

Comment:
Let's also check that the specified `clientSideComponentProperties` are valid JSON